lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3738,16 +3741,17 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
   /// Mapping from original basic blocks to dominated active values.
   DenseMap<SILBasicBlock *, SmallVector<SILValue, 8>> activeValues;
 
-  /// Local adjoint values to be cleaned up. This is populated when adjoint
-  /// emission is run on one basic block and cleaned before processing another
-  /// basic block.
-  SmallVector<AdjointValue, 8> blockLocalAdjointValues;
-
   /// Mapping from original basic blocks and original active values to
   /// corresponding adjoint block arguments.
   DenseMap<std::pair<SILBasicBlock *, SILValue>, SILArgument *>
       activeValueAdjointBBArgumentMap;
 
+  /// Mapping from original basic blocks to local adjoint values to be cleaned
+  /// up. This is populated when adjoint emission is run on one basic block and
+  /// cleaned before processing another basic block.
+  DenseMap<SILBasicBlock *, SmallVector<AdjointValue, 8>>
```

Comment:
Note: changed `blockLocalAdjointValues` to a `DenseMap` for more principled/predictable behavior.
Previously, using `SmallVector` caused bugs when the original entry BB isn't the last visited BB during adjoint generation.